src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -486,6 +578,67 @@ class D3
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void VerifyExecutionOrder_TupleLiteralAndDeconstruction_Conditional()
+        {
+            string source = @"
+using System;
+class C
+{
+    int w { set { Console.WriteLine($""setW""); } }
+    int x { set { Console.WriteLine($""setX""); } }
+    int y { set { Console.WriteLine($""setY""); } }
+    int z { set { Console.WriteLine($""setZ""); } }
+
+    C getHolderForW() { Console.WriteLine(""getHolderforW""); return this; }
+    C getHolderForX() { Console.WriteLine(""getHolderforX""); return this; }
+    C getHolderForY() { Console.WriteLine(""getHolderforY""); return this; }
+    C getHolderForZ() { Console.WriteLine(""getHolderforZ""); return this; }
+
+    static void Main()
+    {
+        C c = new C();
+        bool b = false;
+        (c.getHolderForW().x, (c.getHolderForY().y, c.getHolderForZ().z), c.getHolderForX().x) = b ? default : (new D1(), new D2(), new D3());
```

Comment:
>x [](start = 27, length = 1)

nit: it looks like there was a minor problem in the original test. This should probably be `c.getHolderForW().w` instead of `.x`.